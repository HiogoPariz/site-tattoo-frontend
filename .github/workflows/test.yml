name: Test workflow

on:
  pull_request:
    branches: [staging]
jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - name: Check out
        uses: actions/checkout@v2

      - name: Set up Node 14
        uses: actions/setup-node@v3
        with:
          node-version: '14'
          cache: 'yarn'
          cache-dependency-path: yarn.lock

      - name: Install dependencies
        run: yarn install --frozen-lockfile

      - name: Test with jest
        run: yarn test --ci --maxWorkers=100%
