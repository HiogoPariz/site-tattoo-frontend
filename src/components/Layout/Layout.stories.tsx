import React from 'react'
import { Meta, Story } from '@storybook/react'

import { Layout } from './Layout'

const meta: Meta = {
  title: 'Main/Layout',
  component: Layout
}

export default meta
const Template: Story = args => <Layout {...args} />

export const Default = Template.bind({})
