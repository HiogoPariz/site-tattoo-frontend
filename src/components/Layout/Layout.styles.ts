import devices from '@/styles/breakpoints'
import styled from 'styled-components'

export const LayoutContainer = styled.div`
  display: grid;
  width: 100vw;
  height: 100vh;
  justify-content: stretch;
  align-items: stretch;
  grid-template: 'main' 'footer' 'menu';

  @media ${devices.desktop} {
    grid-template: 'menu' 'main' 'footer';
  }
`

export const MainContainer = styled.div`
  grid-area: main;
  display: flex;
  justify-content: center;
  align-items: center;
`
