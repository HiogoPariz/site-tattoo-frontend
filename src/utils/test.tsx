/* eslint-disable import/export */
import React from 'react'
import { render } from '@testing-library/react'
import theme from '@/styles/theme'
import { ThemeProvider } from 'styled-components'

// Add in any providers here if necessary:
// (ReduxProvider, ThemeProvider, etc)
const Providers: React.FC = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'

export { customRender as render }
