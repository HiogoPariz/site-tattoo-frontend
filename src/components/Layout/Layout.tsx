import React from 'react'

import { Menu } from '@/components/Menu/Menu'
import { Footer, FooterGroup } from '@/components/Footer/Footer'
import { NavLink } from '@/types'
import { LayoutContainer, MainContainer } from './Layout.styles'

const menuLinks: NavLink[] = [
  { label: 'Home', to: '/' },
  { label: 'Teste', to: '/teste' }
]

const footerGroups: FooterGroup[] = [{ name: 'rotas', links: menuLinks }]

export const Layout: React.FC = ({ children }) => {
  return (
    <LayoutContainer>
      <Menu links={menuLinks} />
      <MainContainer>{children}</MainContainer>

      <Footer groups={footerGroups} />
    </LayoutContainer>
  )
}
