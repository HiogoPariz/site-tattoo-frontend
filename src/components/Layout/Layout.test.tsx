import React from 'react'
import { render } from '@/utils/test'
import { Layout } from './Layout'

describe('<Layout />', () => {
  it('should render', () => {
    const { container } = render(<Layout />)
    expect(container.firstChild).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
