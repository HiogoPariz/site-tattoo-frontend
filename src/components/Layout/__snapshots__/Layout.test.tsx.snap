// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  grid-area: menu;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1.5rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 1rem;
  padding: 1.5rem;
}

.c2 {
  border-radius: 1rem;
  height: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 2rem;
}

.c4 {
  border: 0.1rem solid #A75E55;
  -webkit-transition: border 200ms;
  transition: border 200ms;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4:focus,
.c4:active {
  border: 0.1rem solid #6D5063;
}

.c6 {
  display: grid;
  width: 100%;
  background-color: #000000;
  grid-area: footer;
  padding: 1.5rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.5rem;
}

.c0 {
  display: grid;
  width: 100vw;
  height: 100vh;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  grid-template: 'main' 'footer' 'menu';
}

.c5 {
  grid-area: main;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:1200px) {
  .c0 {
    grid-template: 'menu' 'main' 'footer';
  }
}

<div
  class="c0"
>
  <header
    class="c1"
  >
    <div
      class="c2"
    >
      <a
        href="/"
      >
        Logo
      </a>
      <form
        class="c3"
      >
        <label
          class="c4"
        >
          teste
          <input
            class="Menustyles__Input-sc-1pqmg6w-2"
          />
        </label>
        <label
          class="c4"
        >
          teste2
          <input
            class="Menustyles__Input-sc-1pqmg6w-2"
          />
        </label>
      </form>
    </div>
  </header>
  <div
    class="c5"
  />
  <div
    class="c6"
  >
    <ul
      class="c7"
    >
      <h2>
        rotas
      </h2>
      <li
        class="Footerstyles__GroupLink-sc-z3hj90-2"
      >
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="Footerstyles__GroupLink-sc-z3hj90-2"
      >
        <a
          href="/teste"
        >
          Teste
        </a>
      </li>
    </ul>
  </div>
</div>
`;
